### PR TITLE
Implement Lesson-related methods. Fix some small bugs

### PR DESCRIPTION
**Pull request by UncleGrandpa925**
Apart from Integrating Lesson to Model, severals bugs were fixed. These bugs were caused by incomplete integration of new classes to the original code base

### CODE REVIEW

File: src/main/java/seedu/address/model/TrackIter.java [hunk 1]
```diff
@@ -207,16 +249,23 @@ public String toString() {
         return modules.asUnmodifiableObservableList();
     }
 
-    @Override
```
Comment (simonteozw) [inline src/main/java/seedu/address/model/TrackIter.java]: Why is `@Override` removed? The `getTaskList` method is from `ReadOnlyTrackIter` so we should keep the `@Override` right?

**Review by simonteozw — COMMENTED** [review]
Looks good except for 1 comment. Is this the final PR for concrete classes? If yes we should re-enable checkStyle

File: src/main/java/seedu/address/model/TrackIter.java (hunk 1 above)
Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/TrackIter.java]: Wow I think this is a bug happend when I'm copying things

Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/TrackIter.java]: fixed!

Comment (simonteozw) [reply src/main/java/seedu/address/model/TrackIter.java]: Is this the final PR for concrete classes? If yes we should re-enable checkStyle

Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/TrackIter.java]: Ah okay okay let's enable it

Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/TrackIter.java]: @simonteozw Maybe you approve this PR first then I make another PR? I just run the code formatter and it changed like 80 files

**Review by simonteozw — APPROVED** [review]
LGTM

**Pull request merged**